src/async-wrap.cc

```diff
@@ -407,6 +407,47 @@ static void DisablePromiseHook(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+class DestroyParam {
+ public:
+  double asyncId;
+  v8::Persistent<Object> target;
+  v8::Persistent<Object> propBag;
+};
+
+
+void AsyncWrap::WeakCallback(
+  const v8::WeakCallbackInfo<DestroyParam>& info
+) {
+  HandleScope scope(info.GetIsolate());
+
+  Environment* env = Environment::GetCurrent(info.GetIsolate());
+  DestroyParam* p = info.GetParameter();
+  Local<Object> propBag = PersistentToLocal(info.GetIsolate(), p->propBag);
```

Comment:
Tiny nit: `prop_bag` for local variables (See https://github.com/nodejs/node/blob/master/CPP_STYLE_GUIDE.md :) )